src/backend/distributed/sql/udfs/get_missing_time_partition_ranges/latest.sql

```diff
@@ -0,0 +1,221 @@
+             datetime_string_format := datetime_string_format || 'SS';
+        END IF;
+    END IF;
+
+    WHILE current_range_from_value < to_value LOOP
+        /*
+         * Check whether partition with given range has already been created
+         * Since partition interval can be given with different types, we are converting
+         * all variables to timestamptz to make sure that we are comparing same type of parameters
+         */
+        PERFORM * FROM pg_catalog.time_partitions tp
+        WHERE
+            tp.from_value::timestamptz = current_range_from_value::timestamptz AND
+            tp.to_value::timestamptz = current_range_to_value::timestamptz AND
+            parent_table = table_name;
+        IF found THEN
+            current_range_from_value := current_range_to_value;
+            current_range_to_value := current_range_to_value + partition_interval;
+            CONTINUE;
+        END IF;
+
+        /*
+         * Check whether any other partition covers from_value or to_value
+         * That means some partitions doesn't align with the initial partition.
+         * In other words, gap(s) exist between partitions which is not multiple of intervals.
+         */
+        SELECT tp.from_value::text, tp.to_value::text
+        INTO manual_partition_from_value_text, manual_partition_to_value_text
+        FROM pg_catalog.time_partitions tp
+        WHERE
+            ((current_range_from_value::timestamptz >= tp.from_value::timestamptz AND current_range_from_value < tp.to_value::timestamptz) OR
+            (current_range_to_value::timestamptz > tp.from_value::timestamptz AND current_range_to_value::timestamptz < tp.to_value::timestamptz)) AND
+            parent_table = table_name;
+
+        IF found THEN
+            RAISE 'Partition with the range from % to % does not align with the initial partition. Please make sure that no gap(s) exists between existing partitions', -- TODO: Check the message
+            manual_partition_from_value_text,
+            manual_partition_to_value_text;
+        END IF;
+
+        IF partition_column_type = 'date'::regtype THEN
+            SELECT current_range_from_value::date::text INTO current_range_from_value_text;
+            SELECT current_range_to_value::date::text INTO current_range_to_value_text;
+        ELSIF partition_column_type = 'timestamp without time zone'::regtype THEN
+            SELECT current_range_from_value::timestamp::text INTO current_range_from_value_text;
+            SELECT current_range_to_value::timestamp::text INTO current_range_to_value_text;
+        ELSIF partition_column_type = 'timestamp with time zone'::regtype THEN
+            SELECT current_range_from_value::timestamptz::text INTO current_range_from_value_text;
+            SELECT current_range_to_value::timestamptz::text INTO current_range_to_value_text;
+        ELSE
+            RAISE 'type of the partition column of the table % must be date, timestamp or timestamptz', table_name;
```

Comment:
should we maybe do this part of the check a little earlier instead of trying to cast the to_value to timestamptz and catching the invalid_datetime_format?

Reply:
We already check this earlier, but left this here as a kind of assert.